app/templates/js/index.jsx

```diff
@@ -1,6 +1,6 @@
 /*** @jsx React.DOM */
 
 var React = require('react'),
-    App = require('./components/App');
+    App = require('./components/App.jsx');
```

Comment:
It does if your file extension is not `.js`. The React convention is to use `.jsx` as the extension.